app/design/adminhtml/default/default/template/widget/grid/massaction.phtml

```diff
@@ -48,12 +48,24 @@
                     <?php echo $this->getBlockHtml('formkey')?>
                     <fieldset>
                         <span class="field-row">
-                            <label><?php echo $this->__('Actions') ?></label>
+                            <label for="<?php echo $this->getHtmlId() ?>-select"><?php echo $this->__('Actions') ?></label>
                             <select id="<?php echo $this->getHtmlId() ?>-select" class="required-entry select absolute-advice local-validation">
                                 <option value=""></option>
-                                <?php foreach($this->getItems() as $_item): ?>
-                                    <option value="<?php echo $_item->getId() ?>"<?php echo ($_item->getSelected() ? ' selected="selected"' : '')?>><?php echo $_item->getLabel() ?></option>
-                                <?php endforeach ?>
+                                <?php foreach($this->getGroupedItems() as $key => $group): ?>
+                                    <?php if ($key === 'default'): ?>
+                                        <?php foreach($group as $_item): ?>
+                                            <option value="<?php echo $_item->getId() ?>"<?php echo ($_item->getSelected() ? ' selected="selected"' : '')?>><?php echo $_item->getLabel() ?></option>
+                                        <?php endforeach; ?>
+                                    <?php elseif ($key === 'grouped'): ?>
+                                        <?php foreach($group as $label => $_massGroup): ?>
+                                            <optgroup label="<?php echo $this->quoteEscape($label); ?>">
+                                                <?php foreach($_massGroup as $_item): ?>
+                                                    <option value="<?php echo $_item->getId() ?>"<?php echo ($_item->getSelected() ? ' selected="selected"' : '')?>><?php echo $_item->getLabel() ?></option>
+                                                <?php endforeach; ?>
+                                            </optgroup>
+                                        <?php endforeach; ?>
+                                    <?php endif; ?>
+                                <?php endforeach; ?>
```

Comment:
Please remove the semicolon after every endif, endforeach. See on the left side how it was initially.

Reply:
@ADDISON74 fixed :-)